lib/pure/ospaths.nim

```diff
@@ -423,12 +423,22 @@ proc isAbsolute*(path: string): bool {.rtl, noSideEffect, extern: "nos$1".} =
   ## Checks whether a given `path` is absolute.
   ##
   ## On Windows, network paths are considered absolute too.
+  runnableExamples:
+    doAssert "".isAbsolute.not
+    doAssert nil.isAbsolute.not
+    doAssert ".".isAbsolute.not
+    when defined(posix):
+      doAssert "/".isAbsolute
+      doAssert "a/".isAbsolute.not
+
+  if len(path) == 0: return false
+
   when doslikeFileSystem:
     var len = len(path)
-    result = (len > 0 and path[0] in {'/', '\\'}) or
+    result = (path[0] in {'/', '\\'}) or
               (len > 1 and path[0] in {'a'..'z', 'A'..'Z'} and path[1] == ':')
   elif defined(macos):
-    result = path.len > 0 and path[0] != ':'
+    result = path[0] != ':'
```

Comment:
This needs to be ``path.len == 0 or path[0] != ':'``, maybe.

Reply:
* actually `or` is not the right thing since empty path should not be absolute
* however I'm already checking for `len=0` at function top-level
* just found another preexisting bug which my latest commit fixes: the condition here should be negated! ":" would be absolute on macos (given that ":" meant the new "/" separator); on that note, see my post https://forum.nim-lang.org/t/4039 "should we keep supporting macos? (as opposed to macosx)"